File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    private boolean check(int i,int[] arr,List<Integer> list) {
        for(int k = 0; k < arr.length; k++) {
            if(i == arr[k]) return false;
            if(i < arr[k]) {
                list.add(i);
                return true;
            }
        }
        return true;
    }
    public int findKthPositive(int[] arr, int k) {
        ArrayList<Integer> list = new ArrayList<>();
        int count = 0;
        for(int i = 1; i <= arr[arr.length - 1]; i++) {
            if(check(i,arr,list)) {
                count++;
            }
            if(k == count) return list.get(list.size() - 1);
        }
        count = k - count;
        return arr[arr.length - 1] + count;
    }
}

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int n = arr.size();
        int low = 0 , high = n - 1;
        while(low <= high) {
            int mid = (low + high) / 2;
             int missing = arr[mid] - (mid + 1);
            if(missing < k) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        
        // int missing = arr[high] - high - 1;
        // int more_req = k - missing;
        // return (arr[high] + more_req);
        
        
        return high + k +1;
    }
};